frontend/src/components/header.js

```diff
@@ -72,7 +73,11 @@ export function Header(props) {
 
     // filters the routes to include only the current route, then maps all of that route's subroutes to BootstrapNavItems
     const availableSubroutes = routes
-        .filter((route) => route.route.substring(1) === mainRoute)
+        .filter(
+            (route) =>
+                route.route.substring(1) === mainRoute ||
+                fullRoute.startsWith(route.route)
```

Comment:
This is unrelated to the current PR, but it would be nice to have a dynamic header component, where matching rules would either come as props or be defined differently, because right now we are stuck with routes `/42/assignments/...` instead of `/positions/42/assignments` for that exact reason

Reply:
This change already fixes it. I changed all the routes to `/positions/42/assignments`